### ISSUE TITLE
Automatic damage given to opponent by just clicking on 'study' for a deck

### ISSUE THREAD

**Issue by CaptainDead** [Unlucky-Life/ankimon#60]
The addons considers just clicking on 'study' for a deck the same as completing a card or clicking on 'good' for a card, I get a pop up of getting a damage just by clicking on 'study' and the opponents's health is decreased.
![image](https://github.com/Unlucky-Life/ankimon/assets/139457914/18296260-0fb6-409f-a164-2984cae84c6b)

If possible it would be nice if that doesn't happen and damage is only given when we finish cards.
Thanks for the addon🙏

**Comment by CaptainDead** [comment]
Also like if I get a pop up saying, '1 damage dealt' then the damage is only shown in 'ankimon window' while the healthbar of the opponent stays the same, that is they are not in complete sync.
![image](https://github.com/Unlucky-Life/ankimon/assets/139457914/8ed096d1-9525-448e-8738-9d1890f7e50e)

**Comment by CaptainDead** [comment]
and btw how does the pokemone's level increase? I don't think there has been any increase in the XP of my main pokemone

**Comment by nurkittel** [comment]
Same if you press undo. Its like answering a card.

**Comment by CaptainDead** [comment]
yeah clicking study, skip, undo, and clicking on good, hard etc... all of them gives damage, so it would be better if just clicking on good, hard,etc would give damage

**Comment by CaptainDead** [comment]
and does our health ever decrease? cuz then there is smwhat no point of health, so it would be great if our health also decrease, like when we press 'again' or 'hard' button, more decrease in our health again and less decrease in hard...

**Comment by Unlucky-Life** [comment]
@CaptainDead Yes ! your mainpokemon level increases by gaining xp. Every time you defeat a wild pokemon, your main pokemon gains xp !

**Comment by CaptainDead** [comment]
somehow my main pokemon's xp never increases even though I have defeated some wild ones and btw the xp bar in ankimon window, does that also increase as we fight? or its just for show for now
![image](https://github.com/Unlucky-Life/ankimon/assets/139457914/6b322b4f-f939-40d9-bd4e-b59640a22e62)

**Comment by Unlucky-Life** [comment]
@CaptainDead The xp bar does increase aswell in reviewer.
Thats weird tho, that you are not gaining any xp.
![image](https://github.com/Unlucky-Life/ankimon/assets/77027147/4160d57d-daf9-4e22-84e0-4cc73cc15ebf)

**Comment by Unlucky-Life** [comment]
@CaptainDead i tested the fix posted in another issue and this should get rid of the issue with the pressing!

**Comment by CaptainDead** [comment]
[mainpokemon.json](https://github.com/Unlucky-Life/ankimon/files/14970974/mainpokemon.json)
[mypokemon.json](https://github.com/Unlucky-Life/ankimon/files/14970975/mypokemon.json)
I shared the files u asked for.

I did restart it, and have restarted my anki a lot of times since then as I've been using ur addon for a few days now, so idk why the XP doesn't increase.
Charmander is my main.
![image](https://github.com/Unlucky-Life/ankimon/assets/139457914/bba5f518-3f62-4489-b195-b5a9723ce37d)

**Comment by Unlucky-Life** [comment]
![image](https://github.com/Unlucky-Life/ankimon/assets/77027147/c8af16ea-ddd5-4e40-aa3e-3e182184cdc6)
@CaptainDead => thats quite weird - i tested it out and it worked fine.
Do you have anki running as admin ?

**Comment by CaptainDead** [comment]
No its not running as admin, why is it happening with just me? can u think of a way to fix this?

**Comment by Unlucky-Life** [comment]
@CaptainDead 
I might have a idea, try these two methods:
1. Run anki as admin => (you maybe are not allowing anki to rewrite files - which basically is not allowing the saving of xp and other data)
2. Could you make a screenshot of your configurations and addons ? (maybe there is a glith or conflict with another addon)
3. try by using the newest ankimon version - run this installer:
https://github.com/Unlucky-Life/ankimon/releases/download/1.269/ankimon-1.269-anki21-ankiweb.ankiaddon

**Comment by CaptainDead** [comment]
Step 1 - Did not work.
Step 2 - Add on error 
![image](https://github.com/Unlucky-Life/ankimon/assets/139457914/723b8427-d02e-4981-a95c-4ae8d7ab6824)
Debug info - 
Anki 23.12.1 (1a1d4d54)  (ao)
Python 3.9.15 Qt 6.6.1 PyQt 6.6.1
Platform: Windows-10-10.0.22621

When loading Ankimon:
Traceback (most recent call last):
  File "aqt.addons", line 245, in loadAddons
  File "C:\Users\capta\AppData\Roaming\Anki2\addons21\1908235722\__init__.py", line 114, in <module>
    with open(sound_list_path, 'r') as json_file:
FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\capta\\AppData\\Roaming\\Anki2\\addons21\\1908235722\\addon_files\\sound_list.json'


Screenshot of addons -
![image](https://github.com/Unlucky-Life/ankimon/assets/139457914/8f158e00-462e-47d2-b6ef-e2706b7117f6)
![image](https://github.com/Unlucky-Life/ankimon/assets/139457914/83fb16f1-4d3b-4778-ac02-79c06ffc462a)

Screenshot of Ankimon's Config - 
![image](https://github.com/Unlucky-Life/ankimon/assets/139457914/1d34fd78-483a-4261-88de-7343aafb7e01)

I hope the next update automatically fix this problem smhw

**Comment by Unlucky-Life** [comment]
@CaptainDead I sorry i forgot some files !
Try with this installer again: https://github.com/Unlucky-Life/ankimon/releases/download/1.269/ankimon-1.269-anki21-ankiweb.ankiaddon

**Comment by CaptainDead** [comment]
ahh sry, getting error again
![image](https://github.com/Unlucky-Life/ankimon/assets/139457914/f51897bf-b657-4fef-801c-79a6bac3a7e8)

Debug info - 

Anki 23.12.1 (1a1d4d54)  (ao)
Python 3.9.15 Qt 6.6.1 PyQt 6.6.1
Platform: Windows-10-10.0.22621

When loading Ankimon:
Traceback (most recent call last):
  File "aqt.addons", line 245, in loadAddons
  File "C:\Users\capta\AppData\Roaming\Anki2\addons21\1908235722\__init__.py", line 237, in <module>
    reviewer_text_message_box_time = config["reviewer_text_message_box_time"] #time in seconds for text message
KeyError: 'reviewer_text_message_box_time'

**Comment by Unlucky-Life** [comment]
[config.json](https://github.com/Unlucky-Life/ankimon/files/14971771/config.json)
Ah man, im so sorry !
Another file i have missed !
Place this in your addon folder - and then try opening anki again

**Comment by CaptainDead** [comment]
Ahh man got the error again smhw


JOKING
well the addon worked. thanks for the efforts but my problem wasn't solved, could a complete reinstall solve this? like deleting the whole folder and then reinstalling the addon and downloading all the files as we do at the start.

Btw I love the new health bar,  it looks great.

![image](https://github.com/Unlucky-Life/ankimon/assets/139457914/a7bd5e7e-ac20-4499-8811-8f2af7029cb1)

**Comment by Unlucky-Life** [comment]
@CaptainDead 
Ah rip. Can you make a video of you defeating a pokemon ?
(This might help the process of elimination to see the issue)

A complete reinstall wont work i think (atleast i dont see a reason for it to be working then). 
But you may aswell try anyway.

**Comment by CaptainDead** [comment]
tf it doesn't make any sense😭 I was making the video and It worked smhw this time😭
I didn't even do anything, but last night it didn't work and now its working in the morning...

Now I understand why 'XP' and 'Next Level' was written at the top corners.
![image](https://github.com/Unlucky-Life/ankimon/assets/139457914/fa2a5bad-38b9-4f8e-bb96-911d7d3e7230)

Btw is the xp bar in Reviewer and xp bar in Ankimon window supposed to be filling as we fight or only after the opponent is defeated? cuz mine only fills up after I have completely defeated a pokemon.
I get this pop up after defeating them -
![image](https://github.com/Unlucky-Life/ankimon/assets/139457914/544bcf79-8ead-486b-a2f6-cc9265acf228)

and btw it would be nice if we know how much more xp we need for next level, like here it could  54/60 instead of just 54 etc, just a suggestion
![image](https://github.com/Unlucky-Life/ankimon/assets/139457914/6ef5526b-0842-429a-a5e3-738db9dce99f)

Anyways its working now, thanks for the help🙏

**Comment by CaptainDead** [comment]
I really love the bars now, one at top one at the bottom, this addon rocks, its so underrated right now fr
and hehe  ig I am the first guy to try the latest version? as u haven't uploaded it yet

**Comment by CaptainDead** [comment]
![image](https://github.com/Unlucky-Life/ankimon/assets/139457914/c29851df-839e-466c-89aa-ec8db0a9409b)
Btw what is the use of attacks that does no damage? like will they do something in future? cuz then those attacks are some what of a waste and do no damage thus it takes longer to defeat the opponent, just sharing my thoughts.

**Comment by CaptainDead** [comment]
btw u can use https://pixeldrain.com/ to share files as well, it lets u share upto 20GB for free if I remember it right.

**Comment by CaptainDead** [comment]
Suggestion - I think the health count should be near the individual pokemone's names, it looks good where it is when the health is full but once the health decreases, it looks out of place.
![image](https://github.com/Unlucky-Life/ankimon/assets/139457914/2bfc8fd1-14a0-440f-8032-018fdc61b74f)

**Comment by Unlucky-Life** [comment]
@CaptainDead so yes attacks that do status effects do work at the moment
So it actually decreases damage, reduces defense etc

**Comment by Unlucky-Life** [comment]
@CaptainDead 
![image](https://github.com/Unlucky-Life/ankimon/assets/77027147/6ebcc633-3b70-47e1-b53a-0ca51582ec10)
also in the show collection => if xp bar reaches right corner = level up

**Comment by Unlucky-Life** [comment]
@CaptainDead So basically:
You need to defeat a enemy pokemon completely either by not having the ankimon window open:
Then after it says defeat and you do another round the enemy pokemon is defeated and your mainpokemon gains xp

Or you have ankimon window open: Then you need to select either defeat or catch pokemon to have another pokemon spawn
